tensordict/prototype/tensorclass.py

```diff
@@ -397,26 +442,22 @@ def _getattr(self, attr: str) -> Any:
         return res
     func = res
 
-    @functools.wraps(getattr)
+    @functools.wraps(func)
     def wrapped_func(*args, **kwargs):
         res = func(*args, **kwargs)
-        # Handling nested tensor class
-        non_tensor_dict = {}
-        for key, value in self._non_tensordict.items():
-            if is_tensorclass(value):
-                temp = getattr(value, attr)
-                if callable(temp):
-                    # Recursively calling for nested tensor classes
-                    non_tensor_dict[key] = temp(*args, **kwargs)
-                else:
-                    non_tensor_dict[key] = temp
-            else:
-                non_tensor_dict[key] = value
         if isinstance(res, TensorDictBase):
-            new = self._from_tensordict(res, non_tensor_dict)
-            return new
-        else:
-            return res
+            if attr.endswith("_"):
+                # in-place operation, return the current object
+                return self
```

Comment:
I think this is fine? If we ever added a method with a trailing underscore that wasn't an in-place operation it could cause some problems, but that would already be a big break from convention

Reply:
Yep I can already see a couple
`lock`
`unlock`
`pin_memory` IIRC
We'd need to edit these methods to match the `_` trailing convention